src/Zonotope.jl

```diff
@@ -319,3 +321,66 @@ function scale(α::Real, Z::Zonotope)
     gi = α .* Z.generators
     return Zonotope(c, gi)
 end
+
+"""
+    ngens(Z::Zonotope)::Int
+
+Return the number of generators of a zonotope.
+
+### Input
+
+- `Z` -- zonotope
+
+### Output
+
+Integer representing the number of generators.
+"""
+ngens(Z::Zonotope)::Int = size(Z.generators, 2)
+
+"""
+    reduce_order(Z::Zonotope, r::Int)
+
+Reduce the order of a zonotope by overapproximating with a zonotope with less
+generators.
+
+### Input
+
+- `Z` -- zonotope
+- `r` -- desired order
+
+### Output
+
+A new zonotope with less generators, if possible.
+
+### Algorithm
+
+This function implements the algorithm described in A. Girard's
+*Reachability of Uncertain Linear Systems Using Zonotopes*, HSCC. Vol. 5. 2005.
+"""
+function reduce_order(Z::Zonotope, r::Int)
+    c, G = Z.center, Z.generators
+    p, d = ngens(Z), dim(Z)
+    h = zeros(p)
```

Comment:
`zeros(N, p)` (you need to define `N` first).

Reply:
I checked it: For `Rational` it returns `Float64`, but for `Float32` it returns `Float32`. So let us keep it as is (`Float32` is automatically promoted to `Float64`, I checked it).